hazelcast/include/hazelcast/client/internal/socket/BaseSocket.h

```diff
@@ -90,54 +90,45 @@ namespace hazelcast {
+                        socket_strand_.post([connection, invocation, message, this]() {
                             if (!check_connection(connection, invocation)) {
                                 return;
                             }
 
                             bool success;
                             int64_t message_call_id;
-                            do {
-                                auto call_id = ++call_id_counter_;
-                                struct correlation_id {
-                                    int32_t connnection_id;
-                                    int32_t call_id;
-                                };
-                                union {
-                                    int64_t id;
-                                    correlation_id composed_id;
-                                } c_id_union;
-                                c_id_union.composed_id = {connection->get_connection_id(), call_id};
-                                message_call_id = c_id_union.id;
-                                message->set_correlation_id(c_id_union.id);
-                                success = connection->invocations.insert({message_call_id, invocation}).second;
-                            } while (!success);
-
-                            auto handler = [=](const boost::system::error_code &ec,
-                                               std::size_t bytes_written) {
-                                if (ec) {
-                                    auto message = (boost::format{
-                                            "Error %1% during invocation write for %2% on connection %3%"} %
-                                                    ec % *invocation % *connection).str();
-                                    connection->close(message);
-                                } else {
-                                    // update the connection write time
-                                    connection->last_write_time(std::chrono::steady_clock::now());
-                                }
+                            auto call_id = ++call_id_counter_;
+                            struct correlation_id {
+                                int32_t connnection_id;
+                                int32_t call_id;
                             };
+                            union {
+                                int64_t id;
+                                correlation_id composed_id;
+                            } c_id_union;
+                            c_id_union.composed_id = {connection->get_connection_id(), call_id};
+                            message_call_id = c_id_union.id;
+                            success = connection->invocations.insert({message_call_id, invocation}).second;
+                            if (success) {
+                                message->set_correlation_id(c_id_union.id);
+                            }
+
+                            //message->print_frame_sizes();
```

Comment:
If this is a leftover we can delete this

Reply:
yes, leftover. Removed.